source/lambda/use-case-management/model/use-case-validator.ts

```diff
@@ -59,12 +62,57 @@ export class UseCaseValidator {
             useCase.configuration.LlmParams!.PromptParams.DisambiguationPromptTemplate = modelInfo.DisambiguationPrompt;
         }
 
+
+        if (useCase.configuration.AuthenticationParams) {
+            switch (useCase.configuration.AuthenticationParams.AuthenticationProvider) {
+                case AUTHENTICATION_PROVIDERS.COGNITO:
+                    // overriding the previously set CognitoDomainPrefix parameter 
+                    // by fetching it dynamically based on the set user pool
+
+                    const existingUserPoolId = useCase.cfnParameters?.get(CfnParameterKeys.ExistingCognitoUserPoolId);
+                    if (!existingUserPoolId) {
+                        throw new Error(`Undefined user pool provided for the cognito authentication provider.`)
+                    }
+
+                    const cognitoDomainPrefix = await this.getCognitoDomainPrefixByUserPool(existingUserPoolId);
+
+                    if (!useCase.cfnParameters) {
+                        throw new Error(`CFNParameters are not available yet for setting Cognito Domain Prefix.`)
+                    }
+
+                    useCase.cfnParameters.set(CfnParameterKeys.CognitoDomainPrefix, cognitoDomainPrefix);
+
+                    break;
+            }
+        }
+
         await UseCaseValidator.checkModelInputPayloadSchema(useCase); // NOSONAR - typescript:S4123 - await is required in tests despite seeming unnecessary
         await UseCaseValidator.checkPromptsAreCompatible(useCase); // NOSONAR - typescript:S4123 - await is required in tests despite seeming unnecessary
         await UseCaseValidator.checkKnowledgeBaseTypeMatchesParams(useCase); // NOSONAR - typescript:S4123 - await is required in tests despite seeming unnecessary
         return useCase;
     }
 
+    private async getCognitoDomainPrefixByUserPool(userPoolId: string) {
+
+        const region = process.env.AWS_REGION;
+
+        const client = new CognitoIdentityProviderClient({ region });
```

Comment:
For aws clients, we provide a helper function to retrieve the correct config (including region and a custom user agent string used for monitoring) as part of the `aws-node-user-agent-config` lambda layer. 
For an example, it is used here: https://github.com/aws-solutions/generative-ai-application-builder-on-aws/blob/1c54a1b33fde54a7dafb99e30c911562ac92b416/source/lambda/use-case-management/cfn/stack-management.ts#L66

Please use this when initializing clients.